Client/qtTeamTalk/mainwindow.cpp

```diff
@@ -4549,6 +4551,86 @@ void MainWindow::slotChannelsSpeakChannelInformationGrid(bool /*checked =false *
     slotUsersSpeakUserInformation(TT_GetMyChannelID(ttInst));
 }
 
+void MainWindow::slotChannelsSpeakChannelStatisticsGrid(bool /*checked =false */)
+{
+    slotChannelsSpeakChannelStatistics();
+}
+
+void MainWindow::slotChannelsSpeakChannelStatistics()
+{
+    QString speakList, voice, mediafile, video, desktop;
+    QVector<int> users = ui.channelsWidget->getUsersInChannel(TT_GetMyChannelID(ttInst));
+    QVector<QString> voice1, mediafile1, video1, desktop1;
+    for (int i=0;i<users.size();i++)
+    {
+        User user = {};
+        TT_GetUser(ttInst, users[i], &user);

```

Comment:
Use ui.channelsWidget->getUser() instead of TT_GetUser(). TT_GetUser() runs in a different thread, so it may yield different results.